### PR TITLE
Faraday

### PR DESCRIPTION
**Pull request by apurvis**
@jarthod 
use faraday instead of httparty

supersedes/includes https://github.com/RipTheJacker/survey-gizmo-ruby/pull/51

not 100% tested yet, but pretty much (like 95%+ tested)

### CODE REVIEW

File: spec/configuration_spec.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@
   end
 
   after(:each) do
-    SurveyGizmo.reset!
+#    SurveyGizmo.reset!
```
Comment (jarthod) [inline spec/configuration_spec.rb]: Are these specs still supposed to be commented ?

File: survey-gizmo-ruby.gemspec [hunk 2]
```diff
@@ -16,8 +16,10 @@ Gem::Specification.new do |gem|
   gem.required_ruby_version = '>= 1.9'
 
   gem.add_dependency 'activesupport', '>= 3.0'
-  gem.add_dependency 'addressable'
-  gem.add_dependency 'awesome_print'
+  gem.add_dependency 'addressable', '~> 2'
+  gem.add_dependency 'awesome_print', '~> 1'
+  gem.add_dependency 'faraday', '~> 0.9'
+  gem.add_dependency 'faraday_middleware'
   gem.add_dependency 'httparty'
```
Comment (jarthod) [inline survey-gizmo-ruby.gemspec]: Still Partying ?

**Comment by jarthod** [comment]
Oh my god you're on fire :fire: 
Don't you want write access to the repo ? ^^

**Comment by jarthod** [comment]
Merry Christmas BTW :christmas_tree:

**Comment by apurvis** [comment]
sure write access would be great.   

i'm going to step away for the holidays.  don't merge this branch until i run more tests though https://github.com/RipTheJacker/survey-gizmo-ruby/pull/51 is totally good to go.

**Comment by apurvis** [comment]
The biggest change is that now instead of parsing and retrying etc with `RestResponse`, there's just layers of faraday middleware that handle the decomposition from text->json->reasonableness.

this is well tested but give it til after christmas just so i can let it chug along for a few days; sometimes edge cases crop up

**Comment by apurvis** [comment]
other changes from #51 - 

1. server_has_new_results_since? - bug fix with enumerables (added spec)
2. Survey#responses method replaces having to call SurveyGizmo::API::Response.all(survey_id: survey.id)
2. better docs
3. Answer now has a submitted_at attribute
4. default number of retries is 1, with a 60 second timeout (enough to recover from a rate limit problem)
5. minor refactor

**Pull request closed without merging**